File: 0645-set-mismatch/0645-set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        unordered_set<int>st(nums.begin(), nums.end());
        
        int n = nums.size();
        
        int expectedSum = (n * (n + 1)) / 2;
        
        int uniqueSum = 0, arraySum = 0;
        
        for(int i : nums)
            arraySum += i;
        
        for(auto it : st)
            uniqueSum += it;
        
        int missing = expectedSum - uniqueSum;
        int duplicate = arraySum - uniqueSum;
        
        return {duplicate, missing};
    }
};

File: 0645-set-mismatch/0645-set-mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int n = nums.length;
        
        int []count = new int[n + 1];
        
        for(int i : nums)
            ++count[i];
        
        int missing = -1, repeat = -1;
        
        for(int i = 1; i < n + 1; i++){
            if(count[i] == 2)   repeat = i;
            if(count[i] == 0)   missing = i; 
        }
        
        return new int[]{repeat, missing};
    }
}